RecoTracker/ExoRegionalTracking/plugins/ExoRegionProducer.cc

```diff
@@ -0,0 +1,231 @@
+#include "RecoTracker/ExoRegionalTracking/plugins/ExoRegionProducer.h"
+
+ExoRegionProducer::ExoRegionProducer(const edm::ParameterSet &cfg, const ExoRegionTFCache *cache)
+    : rParam_(cfg.getParameter<double>("rParam")),
+
+      minRadius_(cfg.getParameter<double>("minRadius")),
+      discriminatorCut_(cfg.getParameter<double>("discriminatorCut")),
+      genMatch_(cfg.getParameter<string>("genMatch")),
+      input_names_(cfg.getParameter<vector<string> >("input_names")),
+      output_names_(cfg.getParameter<vector<string> >("output_names")) {
+  beamSpotToken_ = consumes<reco::BeamSpot>(cfg.getParameter<edm::InputTag>("beamSpot"));
+  trackClustersToken_ =
+      consumes<edm::View<reco::VertexCompositeCandidate> >(cfg.getParameter<edm::InputTag>("trackClusters"));
+  if (genMatch_ != "None")
+    genParticlesToken_ = consumes<edm::View<reco::GenParticle> >(cfg.getParameter<edm::InputTag>("genParticles"));
```

Comment:
shouldn't you use the `mayConsume` ?

Reply:
> shouldn't you use the `mayConsume` ?

No. The code below gets `genParticlesToken_` under the same condition as the consumes declaration here, so `consumes()` is correct.

(and in practice `mayConsume()` currently does the same and `consumes()`, and to make it really functional could require interface changes, possibly similar to EventSetup `mayConsume()`).

Reply:
Obviated by removing `genMatch_`.